### ISSUE TITLE
[BUG] Dates are not correct when loading files saved in "Strict Mode"

### ISSUE THREAD

**Issue by jec006** [exceljs/exceljs#2695]
## 🐛 Bug Report

TL;DR When parsing Strict mode xlsx files, date cells are parsed incorrectly resulting in dates in 1904/1905.

If you go into excel, do "Save As" and select "Strict Open XML Spreadsheet (.xlsx)" it creates a file in the Strict format (rather than the default "Transitional" format).  The strict format has a few differences, but the one I'm running into issues with is the Date formatting.

I found this reference regarding the date formatting: https://www.loc.gov/preservation/digital/formats/fdd/fdd000401.shtml

> Textual and serial numeric representations of dates and times: In the Strict variant of XLSX, dates may be stored using textual forms specified in the ISO 8601 standard. Late in the ISO standardization process for OOXML, a proposal was made to adopt the ISO 8601 format for dates and times in spreadsheets. Dates and times in spreadsheets have usually been stored as numbers (sometimes termed "serial date-time" values)

Unfortunately, when loading a Strict mode xlsx file with excelJS, any date cells are improperly treated as floats and you end up with a date in 1904 or 1905 (depending on OS).

Lib version: 4.4.0

## Steps To Reproduce

Using the attached file: [DateFormatStrict.xlsx](https://github.com/exceljs/exceljs/files/14316248/DateFormatStrict.xlsx)

```javascript
const ExcelJS = require('./excel')
const fs = require('fs/promises')

const fh = await fs.open('../DateFormatStrict.xlsx')
const bufferData = await fh.readFile()

const workbook = new ExcelJS.Workbook();
await workbook.xlsx.load(bufferData)

const cell = workbook.worksheets[0].getCell('A2')

expect(cell.value).to.equal(new Date('02/09/2024'));
// cell.value is actually 1905-07-16T00:00:00.000Z
```

## The expected behaviour:

The date value should be parsed correctly and match what was entered in excel (so a date object representing 02/09/2024).

## Possible solution (optional, but very helpful):

I noticed that the t attribute for the cell values is `d`, so perhaps we could add a handler for that value type to CellXform parseClose here: https://github.com/exceljs/exceljs/blob/master/lib/xlsx/xform/sheet/cell-xform.js#L358

```javascript

else if (this.t === 'd') {
  model.result = new Date(model.value)
}

```
It looks like there is other code that also needs updating since that code is just parsing a float (somewhere else is turning it into a date).  If I have time tonight I'll try to figure it out and make a PR

**Comment by hugo082** [comment]
Hii @jec006 👋 

Any chance to see the PR merged? What are the next steps?
I would be happy to help if needed

_Edit:_ the project seems inactive: https://github.com/exceljs/exceljs/issues/2764

**Comment by suuunly** [comment]
Heard anything back on this issue?
Since not being able to read dates makes this package unusable when it comes to working with dates 😅

**Comment by jec006** [comment]
Hi, I haven't heard anything back and don't have the ability to merge.  I've taken to just using this commit and doing a git reference for this package

**Comment by suuunly** [comment]
It is very unfortunate that they haven't labelled the project as shelved, since a lot of people will hop into using it, without knowing these underlying bugs that will not get fixed 🫠
But thank you for your insight! I will do the same then, and find a different library in future prospects 😁
